python/lsst/meas/modelfit/mixture.cc

```diff
@@ -0,0 +1,217 @@
+    cls.def("getSigma", &MixtureComponent::getSigma);
+    cls.def("setSigma", &MixtureComponent::setSigma);
+    cls.def(
+        "project",
+        (MixtureComponent (MixtureComponent::*)(int) const)&MixtureComponent::project,
+        "dim"_a
+    );
+    cls.def(
+        "project",
+        (MixtureComponent (MixtureComponent::*)(int, int) const)&MixtureComponent::project,
+        "dim1"_a, "dim2"_a
+    );
+    cls.def(py::init<int>(), "dim"_a);
+    cls.def(py::init<Scalar, Vector const &, Matrix const &>(), "weight"_a, "mu"_a, "sigma"_a);
+    cls.def(
+        "__str__",
+        [](MixtureComponent const & self) {
+            std::ostringstream os;
+            os << self;
+            return os.str();
+        }
+    );
+    return cls;
+}
+
+PyMixtureUpdateRestriction declareMixtureUpdateRestriction(py::module & mod) {
+    PyMixtureUpdateRestriction cls(mod, "MixtureUpdateRestriction");
+    cls.def("getDimension", &MixtureUpdateRestriction::getDimension);
+    cls.def(py::init<int>(), "dim"_a);
+    // The rest of this interface isn't usable in Python, and doesn't need to be.
+    return cls;
+}
+
+PyMixture declareMixture(py::module & mod) {
+    afw::table::io::declarePersistableFacade<Mixture>(mod, "Mixture");
+    PyMixture cls(mod, "Mixture");
+    cls.def(
+        "__iter__",
+        [](Mixture & self) { return py::make_iterator(self.begin(), self.end()); },
+        py::keep_alive<0, 1>()
+    );
+    cls.def(
+        "__getitem__",
+        [](Mixture & self, std::ptrdiff_t i) { return self[utils::cppIndex(self.size(), i)]; },
+        py::return_value_policy::reference_internal
+    );
+    cls.def("__len__", &Mixture::size);
+    cls.def("getComponentCount", &Mixture::getComponentCount);
+    cls.def(
+        "project",
+        (std::shared_ptr<Mixture> (Mixture::*)(int) const)&Mixture::project,
+        "dim"_a
+    );
+    cls.def(
+        "project",
+        (std::shared_ptr<Mixture> (Mixture::*)(int, int) const)&Mixture::project,
+        "dim1"_a, "dim2"_a
+    );
+    cls.def("getDimension", &Mixture::getDimension);
+    cls.def("normalize", &Mixture::normalize);
+    cls.def("shift", &Mixture::shift, "dim"_a, "offset"_a);
+    cls.def("clip", &Mixture::clip, "threshold"_a=0.0);
+    cls.def("getDegreesOfFreedom", &Mixture::getDegreesOfFreedom);
+    cls.def("setDegreesOfFreedom", &Mixture::setDegreesOfFreedom,
+                   "df"_a=std::numeric_limits<Scalar>::infinity());
+    cls.def(
+        "evaluate",
+        [](
+            Mixture const & self,
+            MixtureComponent const & component,
+            ndarray::Array<Scalar,1,0> const & array
+        ) -> Scalar {
+            return self.evaluate(component, array.asEigen());
+        },
+        "component"_a, "x"_a
+    );
+    cls.def(
+        "evaluate",
+        [](Mixture const & self, ndarray::Array<Scalar,1,0> const & array) -> Scalar {
+            return self.evaluate(array.asEigen());
```

Comment:
Yes, but that might require a lambda too (since the original C++ signature takes a CRTP base class), and it would require an extra copy.